Terminal.Gui/Views/Scroll/Scroll.cs

```diff
@@ -0,0 +1,203 @@
+﻿#nullable enable
+
+using System.ComponentModel;
+
+namespace Terminal.Gui;
+
+/// <summary>
+///     Provides a proportional control for scrolling through content. Used within a <see cref="ScrollBar"/>.
+/// </summary>
+public class Scroll : View
+{
+    /// <inheritdoc/>
+    public Scroll ()
+    {
+        _slider = new (this);
+        Add (_slider);
+
+        WantContinuousButtonPressed = true;
+        CanFocus = false;
+        Orientation = Orientation.Vertical;
+        Width = Dim.Auto (DimAutoStyle.Content, 1);
+        Height = Dim.Auto (DimAutoStyle.Content, 1);
```

Comment:
@tig if a derived view want to constraint the `Width` and the `Height` to avoid the user set inappropriate values, how that can be done, if they aren't virtual?

Reply:
@tig I need your help here, please. I really want to use `Dim.Auto` on this PR but I'm having problems obtained the maximum value. With `DimAutoStyle.Content`, omitted maximum parameter should use the superview `Viewport` size, I think. I'm wrong with this? I confess that I'm really confuse with this. The bellow test is wrong? I also tried with the commented line.

```cs
    [Fact]
    public void DimAutoStyle_Content_On_SubView_Added_To_A_SubView ()
    {
        var supView = new View { Width = 10, Height = 10 };
        var subView = new View { Width = Dim.Auto (DimAutoStyle.Content, 1), Height = Dim.Auto (DimAutoStyle.Content, 1) };
        //var subView = new View { Width = Dim.Auto (DimAutoStyle.Content, 1), Height = Dim.Auto (DimAutoStyle.Content, 1, Fill ()) };
        supView.Add (subView);

        supView.SetRelativeLayout (supView.Frame.Size);
        Assert.Equal (new (10, 10), subView.Frame.Size);
        Assert.Equal (new (0, 0), subView.Frame.Location);
    }

```

Reply:
Sorry, I just saw this comment. Do you stil need help?